redeployContract.js

```diff
@@ -251,51 +381,64 @@ const deployLargeContract = async (web3, target_address, contract_code, contract
         .catch((error) => {
             console.log('Error while trying to destruct initContract: ', error);
         });
-    return proxyAddress;
+    migrationResult.migrationCompleted = true;
+    return migrationResult;
 };
 
 const setValuesOnInitContract = async (target_address, initContract, keys, values) => {
     console.log('keys: ', keys);
     console.log('values: ', values);
+    if (keys.length < 1) return true;
+    let migrated = false;
     await initContract.methods.setValue(keys, values).send({
         from: target_address,
         gas: 4700000,
         gasPrice: '2000000000'
     })
         .on('error', function (error) {
```

Comment:
Checking for the error event is not enough to catch all errors.

We need to add the .catch() function as well to catch all errors and reliably write the interrupted-migration.json file for continuous migration.

Reply:
Yes